custom_methods.rb

```diff
@@ -0,0 +1,148 @@
+# rubocop:disable Style/CaseEquality
+# rubocop:disable Metrics/ModuleLength
+module Enumerable
+  def change_self
+    return Array self if self.class == Range
+
+    self
+  end
+
+  def my_each
+    my_array = change_self
+    return to_enum unless block_given?
+
+    i = 0
+    while i < my_array.length
+      yield my_array[i]
+      i += 1
+    end
+    my_array
+  end
+
+  def my_each_with_index
+    my_array = change_self
+    return to_enum unless block_given?
+
+    i = 0
+    while i < my_array.length
+      yield my_array[i], i
+      i += 1
+    end
+    my_array
+  end
+
+  def my_select
+    my_array = change_self
+    return to_enum unless block_given?
+
+    arr = []
+    my_array.my_each { |x| arr.push(x) if yield x }
+    arr
+  end
+
+  def my_all?(word = nil)
+    my_array = change_self
```

Comment:
* [ ] should have working my_all? (covered by automated tests) identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2)) 
Let us compare answers in each of the examples
 ``` p %w[ant bear cat].my_all?(/t/)```  != ```%w[ant bear cat].all?(/t/)```
 ``` p [1, 2i, 3.14].my_all?(Numeric)``` != ```[1, 2i, 3.14].all?(Numeric)```